platform-docs/docs/intro.md

```diff
@@ -10,76 +10,105 @@ Let's discover how to use the **Gutenberg Block Editor** to build your own block
+
+```js
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  // highlight-next-line
+  define: { 'process.env.IS_GUTENBERG_PLUGIN': false },
+  plugins: [react()],
+})
+```
+
 ## JSX
 
-We're going to be using JSX to write our UI and components. So one of the first steps we need to do is to configure our build tooling, By default vite supports JSX and and outputs the result as a React pragma. The Block editor uses React so there's no need to configure anything here but if you're using a different bundler/build tool, make sure the JSX transpilation is setup properly.
+We're going to be using JSX to write our UI and components as the block editor is built with React. Using the Vite bootstrap described above there’s no need to configure anything as it outputs the result as a React pragma. If you're using a different bundler/build tool, you may need to configure the JSX transpilation to do the same.
 
 ## Bootstrap your block editor
 
-It's time to render our first block editor.
+It's time to render our first block editor. Update your `src/App.jsx` file with the following code:
 
- - Update your `index.jsx` file with the following code:
 ```jsx
-import { createElement, useState } from "react";
-import { createRoot } from 'react-dom/client';
+import { useState } from "react";
 import {
   BlockEditorProvider,
   BlockCanvas,
 } from "@wordpress/block-editor";
 import { registerCoreBlocks } from "@wordpress/block-library";
+import { getBlockTypes } from "@wordpress/blocks";
 
 // Default styles that are needed for the editor.
-import "@wordpress/components/build-style/style.css";
-import "@wordpress/block-editor/build-style/style.css";
+import componentsStyles from "@wordpress/components/build-style/style.css?raw";
+import blockEditorStyles from "@wordpress/block-editor/build-style/style.css?raw";
+import blockEditorContentStyles from "@wordpress/block-editor/build-style/content.css?raw";
 
 // Default styles that are needed for the core blocks.
-import "@wordpress/block-library/build-style/common.css";
-import "@wordpress/block-library/build-style/style.css";
-import "@wordpress/block-library/build-style/editor.css";
-
-// Register the default core block types.
-registerCoreBlocks();
-
-function Editor() {
+import blocksCommonStyles from "@wordpress/block-library/build-style/common.css?raw";
+import blocksStyles from "@wordpress/block-library/build-style/style.css?raw";
+import blocksEditorStyles from "@wordpress/block-library/build-style/editor.css?raw";
+
+const styles = `
+  ${ componentsStyles }
+  ${ blockEditorStyles }
+  ${ blockEditorContentStyles }
+  ${ blocksCommonStyles }
+  ${ blocksStyles }
+  ${ blocksEditorStyles }
+`;
```

Comment:
While we should do this for "content styles", some of these styles shouldn't be injected into the iframe and others should be injected inside and outside the iframe:

 - components (in both places because of placeholders and buttons)
  - block-editor/style style shouldn't be loaded in the iframe IMO (if something is not working we should fix it in gutenberg), instead it should be loaded in the app directly (so direct import here)

Reply:
Yes, I’d meant to revisit this and get it right. 

> * (if something is not working we should fix it in gutenberg)

Agreed. I can’t quite recall what style issues I’d noted. Perhaps the default block appender was not getting styled but since #66630 that doesn’t appear to be an issue.